main.tf

```diff
@@ -54,19 +56,32 @@ data "aws_iam_policy_document" "security_reporter_lambda_policy_document" {
     ]
   }
 
-  dynamic "statement" {
-    for_each = var.sns_topic_arn != "DUMMY" ? [1] : []
+  statement {
+    effect = "Allow"
 
-    content {
-      effect = "Allow"
-      actions = [
-        "sns:Publish",
-      ]
-      resources = [
-        var.sns_topic_arn,
-      ]
-    }
+    actions = [
+      "ssm:GetParametersByPath"
+    ]
+
+    resources = [
+      "arn:aws:ssm:${data.aws_region.current.name}:${data.aws_caller_identity.current.account_id}:parameter/*",
```

Comment:
I would vote for defining here the full list of parameters the lambda is working with so no access to every, already existing parameters is possible.